src/Compilers/VisualBasic/Portable/Binding/Binder_Lookup.vb

```diff
@@ -1542,6 +1589,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 ExitForFor:
 
                     If ambiguityDiagnostics IsNot Nothing Then
+                        If lookupResult Is Nothing Then
```

Comment:
This is nested inside a check for `lookupResult IsNot Nothing`, should never be true.